File: Ser2023/Company/Company.Web/Modules/Store/CategoriesType/CategoriesTypeRow.cs
using Company.Administration.Entities;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using Store;
using System;
using System.ComponentModel;

namespace Company.Store;

[ConnectionKey("Store"), Module("Store"), TableName("[dbo].[CategoriesType]")]
[DisplayName("Categories Type"), InstanceName("Categories Type")]
[ReadPermission(PermissionKeys.CategoriesType.View)]
[ModifyPermission(PermissionKeys.CategoriesType.Modify)]
[DeletePermission(PermissionKeys.CategoriesType.Delete)]
[LookupScript("Store.CategoriesType", LookupType = typeof(MultiTenantRowLookupScript<>))]
public sealed class CategoriesTypeRow : LoggingRow<CategoriesTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Category Type Id"), Column("CategoryTypeID"), Identity,IdProperty]
    public Int32? CategoryTypeId
    {
        get => fields.CategoryTypeId[this]; 
        set => fields.CategoryTypeId[this] = value; 
    }

    [DisplayName("Type"), NotNull] //, NameProperty]
    public Int16? Type
    {
        get => fields.Type[this]; 
        set => fields.Type[this] = value; 
    }

    [DisplayName("Category Type"), Size(30), NotNull, NameProperty, QuickSearch]
    public String CategoryType
    {
        get => fields.CategoryType[this]; 
        set => fields.CategoryType[this] = value; 
    }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId
    {
        get => fields.TenantId[this]; 
        set => fields.TenantId[this] = value; 
    }

    public Int32Field TenantIdField
    {
        get => fields.TenantId; 
    }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive
    {
        get => fields.IsActive[this]; 
        set => fields.IsActive[this] = value; 
    }

    Int16Field IIsActiveRow.IsActiveField
    {
        get => fields.IsActive; 
    }

    #endregion Tenant & Activ

    public CategoriesTypeRow()
    { 
    }

    public CategoriesTypeRow(RowFields fields)
        : base(fields)
    {
    }

    public class RowFields : LoggingRowFields
    {
        public Int32Field CategoryTypeId;
        public Int16Field Type;
        public StringField CategoryType;
            
        public Int16Field IsActive;
        public Int32Field TenantId;
    }
}

File: Ser2023/Company/Company.Web/Modules/ServerTypes/Store/CategoriesTypeRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface CategoriesTypeRow {
    CategoryTypeId?: number;
    Type?: number;
    CategoryType?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class CategoriesTypeRow {
    static readonly idProperty = 'CategoryTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'CategoryType';
    static readonly localTextPrefix = 'Store.CategoriesType';
    static readonly lookupKey = 'Store.CategoriesType';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CategoriesTypeRow>('Store.CategoriesType') }
    static async getLookupAsync() { return getLookupAsync<CategoriesTypeRow>('Store.CategoriesType') }

    static readonly deletePermission = 'Store:CategoriesType:Delete';
    static readonly insertPermission = 'Store:CategoriesType:Modify';
    static readonly readPermission = 'Store:CategoriesType:View';
    static readonly updatePermission = 'Store:CategoriesType:Modify';

    static readonly Fields = fieldsProxy<CategoriesTypeRow>();
}